app/scripts/controllers/transactions/index.js

```diff
@@ -376,6 +376,22 @@ export default class TransactionController extends EventEmitter {
     return transactions[txId];
   }
 
+  /**
+   * @param {string} txHash - transaction hash
+   * @returns {TransactionMeta} the txMeta who matches the given hash if none found
+   * for the network returns undefined
+   */
+  _getTransactionByHash(txHash) {
+    const { transactions } = this.store.getState();
+    const txByHash = Object.values(transactions).filter((tx) => {
+      if (tx.hash && txHash && tx.hash === txHash) {
+        return tx;
+      }
+      return undefined;
+    });
+    return txByHash[0];
+  }
+
```

Comment:
Instead of doing this, which is potentially 50 iterations of logic for every metric event related to retries, lets modify the `_markNonceDuplicatesDropped` function to, in addition to adding the replacedBy key, adding a replacedById key that records the internal metamask id of the tx that replaced it. This will give us an easier way to quickly look it up.

Reply:
I saw this logic was laid out by @danjm , so tagging him to see if i'm being egregious with my recommendation here.

Reply:
We had a meeting with @danjm today and we agreed with you @brad-decker about this suggestion. Done [here](https://github.com/MetaMask/metamask-extension/pull/15383/commits/1c9525d3f99a0d7e073e759aa96624b20fc8c209).